integration/gemfile_lock_test.go

```diff
@@ -0,0 +1,98 @@
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/paketo-buildpacks/occam"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+	. "github.com/paketo-buildpacks/occam/matchers"
+)
+
+func testGemfileLock(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect     = NewWithT(t).Expect
+		Eventually = NewWithT(t).Eventually
+
+		pack   occam.Pack
+		docker occam.Docker
+	)
+
+	it.Before(func() {
+		pack = occam.NewPack().WithVerbose()
+		docker = occam.NewDocker()
+	})
+
+	context("when the source code contains a Gemfile.lock", func() {
+		var (
+			image     occam.Image
+			container occam.Container
+
+			name   string
+			source string
+		)
+
+		it.Before(func() {
+			var err error
+			name, err = occam.RandomName()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		it.After(func() {
+			Expect(docker.Container.Remove.Execute(container.ID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
+			Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+			Expect(os.RemoveAll(source)).To(Succeed())
+		})
+
+		it("installs the version specified therein", func() {
+			var err error
+			source, err = occam.Source(filepath.Join("testdata", "gemfile_lock_version"))
+			Expect(err).NotTo(HaveOccurred())
+
+			var logs fmt.Stringer
+			image, logs, err = pack.WithNoColor().Build.
+				WithPullPolicy("never").
+				WithBuildpacks(
+					settings.Buildpacks.RVM.Online,
+					settings.Buildpacks.Bundler.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				Execute(name, source)
+			Expect(err).ToNot(HaveOccurred(), logs.String)
+
+			container, err = docker.Container.Run.
+				WithEnv(map[string]string{"PORT": "8080"}).
+				WithPublish("8080").
+				WithPublishAll().
+				Execute(image.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Expected a version of `bundler` which corresponds to `gems` v.3.0.8
+			Eventually(container).Should(BeAvailable())
+			Eventually(container).Should(Serve(ContainSubstring("/layers/com.anynines.buildpacks.rvm/rvm/gems/")).OnPort(8080))
+			Eventually(container).Should(Serve(MatchRegexp(`Bundler version 1\.17\.\d+`)).OnPort(8080))
+
+			Expect(logs).To(ContainLines(
+				MatchRegexp(fmt.Sprintf(`%s \d+\.\d+\.\d+`, settings.Buildpack.Name)),
+			))
+			Expect(logs).To(ContainLines(
+				MatchRegexp(`    Installing RubyGems 3.0.8`),
+				MatchRegexp(`      Successfully built RubyGem`),
+				MatchRegexp(`      Name: bundler`),
+				MatchRegexp(`      Version: 1.17.3`),
```

Comment:
A new version of Bundler will not pass this test. Maybe use a RegEx instead?

Reply:
Rubygems 3.0.8, which is mandatory for a Bundler v1.x.x, installs bundler 1.17.3 after gem update. I suppose it cannot be changed without changing Rubygems v3.0.8 to another version in bundler.go. I cannot say now if there are such plans because I don't know why Gems 3.0.8 are so important for Bundler v1.x.x.